ui/helpers/utils/permission.js

```diff
@@ -6,60 +6,94 @@ import { getRpcCaveatOrigins } from '@metamask/snaps-controllers/dist/snaps/endo
 import { SnapCaveatType } from '@metamask/snaps-utils';
 import { isNonEmptyArray } from '@metamask/controller-utils';
 ///: END:ONLY_INCLUDE_IN
+import classnames from 'classnames';
 import {
   RestrictedMethods,
   ///: BEGIN:ONLY_INCLUDE_IN(flask)
   EndowmentPermissions,
   ///: END:ONLY_INCLUDE_IN
 } from '../../../shared/constants/permissions';
+import Tooltip from '../../components/ui/tooltip';
+import {
+  AvatarIcon,
+  ///: BEGIN:ONLY_INCLUDE_IN(flask)
+  Icon,
+  ///: END:ONLY_INCLUDE_IN
+  ICON_NAMES,
+  ICON_SIZES,
+} from '../../components/component-library';
 ///: BEGIN:ONLY_INCLUDE_IN(flask)
+import { IconColor } from '../constants/design-system';
 import { SNAPS_METADATA } from '../../../shared/constants/snaps';
+import { coinTypeToProtocolName, getSnapDerivationPathName } from './util';
 ///: END:ONLY_INCLUDE_IN
-import { Icon, ICON_NAMES } from '../../components/component-library';
-import { Color } from '../constants/design-system';
-import { coinTypeToProtocolName, getSnapDerivationPathName } from './util'; // eslint-disable-line no-unused-vars
 
 const UNKNOWN_PERMISSION = Symbol('unknown');
 
+///: BEGIN:ONLY_INCLUDE_IN(flask)
+const RIGHT_WARNING_ICON = (
+  <Icon
+    name={ICON_NAMES.DANGER}
+    size={ICON_SIZES.SM}
+    color={IconColor.warningDefault}
+  />
+);
+
+const RIGHT_INFO_ICON = (
+  <Icon
+    name={ICON_NAMES.INFO}
+    size={ICON_SIZES.SM}
+    color={IconColor.iconMuted}
+  />
+);
+///: END:ONLY_INCLUDE_IN
+
+function getLeftIcon(iconName) {
+  return (
+    <AvatarIcon
+      iconName={iconName}
+      size={ICON_SIZES.SM}
+      iconProps={{
+        size: ICON_SIZES.XS,
+      }}
+    />
+  );
+}
+
 const PERMISSION_DESCRIPTIONS = deepFreeze({
   [RestrictedMethods.eth_accounts]: (t) => ({
     label: t('permission_ethereumAccounts'),
-    leftIcon: (
-      <Icon name={ICON_NAMES.EYE} margin={4} color={Color.iconAlternative} />
-    ),
+    leftIcon: getLeftIcon(ICON_NAMES.ETHEREUM),
```

Comment:
Do we have sign-off on changing this icon?

Reply:
Leftover from rebase. I will change it back.